include/components/weapon_component.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef FILE_WEAPON_COMPONENT_HPP
+#define FILE_WEAPON_COMPONENT_HPP
+
+#include <memory>
+#include <optional>
+
+#include "brickengine/components/component_impl.hpp"
+#include "brickengine/components/physics_component.hpp"
+#include "brickengine/components/renderables/texture_component.hpp"
+#include "brickengine/components/data/scale.hpp"
+
+#include "components/damage_component.hpp"
+#include "components/despawn_component.hpp"
+
+class WeaponComponent : public ComponentImpl<WeaponComponent> {
+public:
+    WeaponComponent(DamageComponent bullet_damage, TextureComponent bullet_texture, PhysicsComponent bullet_physics, 
+                    DespawnComponent bullet_despawn, Scale bullet_scale, double fire_rate, std::optional<double> ammo);
+    static std::string getNameStatic();
+
+    // Data
+    // These are the clonable Components/data that a spawned bullet will get.
+    DamageComponent bullet_damage;
+    TextureComponent bullet_texture;
+    // The velocity should ALWAYS be towards the positive directions,
+    // when the weapon is faced in the negative direction, the velocity will get flipped for you
+    PhysicsComponent bullet_physics;
+    DespawnComponent bullet_despawn;
+    Scale bullet_scale;
```

Comment:
This is to make the WeaponComponent as generic as possible. By doing this you basicly describe what a bullet, that the weapon will spawn, will look like. These components will go through the entity manager. [In WeaponSystem](https://github.com/Brick-Studios/BeastArena/blob/66d5ce4d7c477da5a2b24d9a77ee7b03bd89d0d4/lib/systems/weapon_system.cpp#L49) you can see how this plays out. Sadly you cannot make this even more generic by having a vector of components that you will blindly copy, because for instance PhysicsComponent will have to have its velocity's flipped based on player orientation.

Reply:
Thanks for the in-depth explanation